### PR TITLE
An Android instrumentation test using a FlutterView

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @chinmaygarde

File: examples/hello_services/android/app/src/androidTest/java/com/example/flutter/ExampleInstrumentedTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        instr.runOnMainSync(new Runnable() {
+            public void run() {
+                final FlutterView flutterView = (FlutterView) activityRule.getActivity().findViewById(
+                    R.id.flutter_view);
+                flutterView.sendToFlutter("getRandom", message.toString(), new FlutterView.MessageReplyCallback() {
+                    public void onReply(String json) {
+                        try {
+                            JSONObject reply = new JSONObject(json);
+                            random.set(reply.getInt("value"));
+                            latch.countDown();
```
Comment (chinmaygarde) [inline examples/hello_services/android/app/src/androidTest/java/com/example/flutter/ExampleInstrumentedTest.java]: You want to count down in a finally block.

Comment (jason-simmons) [inline examples/hello_services/android/app/src/androidTest/java/com/example/flutter/ExampleInstrumentedTest.java]: done

**Review by chinmaygarde — APPROVED** [review]
lgtm

**Pull request merged**